delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/wiring/AbstractModule.scala

```diff
@@ -0,0 +1,30 @@
+package ch.epfl.bluebrain.nexus.delta.wiring
+
+import akka.persistence.query.{EventEnvelope, Offset}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, Event}
+import izumi.distage.model.definition.ModuleDef
+import monix.bio.UIO
+
+import scala.reflect.ClassTag
+
+// $COVERAGE-OFF$
+abstract class AbstractModule extends ModuleDef {
+
+  /**
+    * Attempts to convert a generic event envelope to a type one.
+    * @param envelope the generic event envelope
+    */
+  def toEnvelope[E <: Event](envelope: EventEnvelope)(implicit Event: ClassTag[E]): UIO[Option[Envelope[E]]] =
```

Comment:
This could go in sourcing no ? So we can use it in tests

Reply:
It can't go in sourcing because the `Envelope` is in the sdk module.